trillian/migrillian/core/util.go

```diff
@@ -15,14 +15,47 @@
 package core
 
 import (
+	"errors"
 	"fmt"
+	"io/ioutil"
 
+	"github.com/gogo/protobuf/proto"
 	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/trillian/migrillian/configpb"
 	"github.com/google/certificate-transparency-go/trillian/util"
 	"github.com/google/certificate-transparency-go/x509"
 	"github.com/google/trillian"
 )
 
+// LoadConfigFromFile reads MigrationConfig from the given filename, which
+// should contain text-protobuf encoded configuration data.
+func LoadConfigFromFile(filename string) (*configpb.MigrationConfig, error) {
+	text, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	var cfg configpb.MigrationConfig
+	if err := proto.UnmarshalText(string(text), &cfg); err != nil {
+		return nil, fmt.Errorf("failed to parse log config: %v", err)
+	}
+	return &cfg, nil
+}
+
+// ValidateConfig verifies that the config is sane.
+func ValidateConfig(cfg *configpb.MigrationConfig) error {
+	switch {
+	case len(cfg.SourceUri) == 0:
+		return errors.New("missing CT log URI")
+	case len(cfg.TrillianUri) == 0:
+		return errors.New("missing Trillian URI")
+	case cfg.LogId <= 0:
+		return errors.New("log ID must be positive")
```

Comment:
Is this true - do we make 63 bit IDs, or just always represent them as uint64?

Reply:
[yes](https://github.com/google/trillian/blob/e856bae55959e54d2e480790d7c881aced7a4b5c/storage/tree_id.go#L23)

Reply:
Excellent :)